localstack-core/localstack/aws/handlers/tracing.py

```diff
@@ -0,0 +1,22 @@
+from localstack.aws.api import RequestContext
+from localstack.aws.chain import Handler, HandlerChain
+from localstack.http import Response
+from localstack.utils.xray.trace_header import TraceHeader
+
+
+class TraceContextParser(Handler):
+    """
+    A handler that parses trace context headers, including:
+    * AWS X-Ray trace header: https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader
+    """
+
+    def __call__(self, chain: HandlerChain, context: RequestContext, response: Response):
+        # TODO: handle case-insensitive headers because HTTP headers are case-insensitive:
+        #  https://stackoverflow.com/a/5259004
+        # TODO: write test case for case-insensitive xray header (verified manually)
+        trace_header_str = context.request.headers.get("X-Amzn-Trace-Id")
```

Comment:
Do we have some efficient CaseInsensitiveDict utility we could use here to retrieve headers?

Reply:
`context.request.headers` (it's the werkzeug Header data structure) should already be case insensitive. so you could also call `.get("X-AMAZN-TrAcE-id")`

Reply:
thanks, it works 🥳